src/Castle.Facilities.AspNetCore/Activators/CrossWiredComponentActivator.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright 2004-2018 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Facilities.AspNetCore.Activators
+{
+	using System.Collections.Generic;
+	using System.Linq;
+
+	using Castle.Core;
+	using Castle.MicroKernel;
+	using Castle.MicroKernel.ComponentActivator;
+	using Castle.MicroKernel.LifecycleConcerns;
+
+	public class CrossWiredComponentActivator : DefaultComponentActivator
+	{
+		public CrossWiredComponentActivator(ComponentModel model, IKernelInternal kernel, ComponentInstanceDelegate onCreation, ComponentInstanceDelegate onDestruction) : base(model, kernel, onCreation, onDestruction)
+		{
+		}
+
+		/// <summary>
+		/// We are overriding disposables to not be tracked, this is because cross-wired components are tracked by implicit scopes that call dispose from ServiceProviders. 
+		/// </summary>
+		/// <param name="steps">All decomission concerns are fine except DisposalConcern</param>
+		/// <param name="instance">The cross wired instance</param>
+		protected override void ApplyConcerns(IEnumerable<IDecommissionConcern> steps, object instance)
+		{
+			if (Model.LifestyleType == LifestyleType.Scoped || Model.LifestyleType == LifestyleType.Singleton)
+			{
+				foreach (var concern in steps.Where(x => x.GetType() != typeof(DisposalConcern)))
+				{
+					concern.Apply(Model, instance);
```

Comment:
I assume you weren't able to just remove the `DisposalConcern` from `ComponentModel.Lifecycle` in `CrossWiringComponentModelContributor`?

Will `LateBoundDecommissionConcerns` also cause the same problem? I guess those can't really be used with this facility as it needs to know the component type upfront to cross wire it.

Reply:
We are cool with perf testing here: https://github.com/fir3pho3nixx/Windsor.AspNetCore/commit/815221ed20cd9388fc7f89a38a04b4642163c511

Reply:
Fuck sake. We have a memory leak. 

Just watched it grow from 220mb to 350mb using typed factories. Che cristo.

Reply:
@jonorossi it is not fair I eat this on this PR. I have been quite vocal about typed factories. They are rubbish for many reasons.  Is it fair to say I do not support typed factories in this PR? Can I sort that next? Not cool that I get held to account for ex-emeritus.

Reply:
Please see: https://github.com/castleproject/Windsor/issues/410

My confidence is knocked.

Reply:
> We have a memory leak.
>
> it is not fair I eat this on this PR. I have been quite vocal about typed factories. They are rubbish for many reasons. Is it fair to say I do not support typed factories in this PR? Can I sort that next? Not cool that I get held to account for ex-emeritus.
>
> ** Where are you guys anyway, help now right or I will re-write your shit **

I'm not sure exactly why you are getting so aggressive here, I get that this pull request has gone on a while. Just a polite reminder the only active committers are those listed on the team page, there should be no expectation of others contributing.

That aside, we can definitely merge this not supporting typed factories, we can even release this without supporting typed factories if you want, remember we get to choose the direction and releasing software with known defects or limitations isn't the end of the world.

Reply:
Thanks @jonorossi. I still believe we need to support typed factories but I know from our issue tracker that comes with even longer tail. 

As long as we know what the problems are before we release I am good. 

Aside, we had our first use case for a cross wire today and I dropped this PR in and it just worked! Very happy with the current state of this.

Reply:
The memory leak is small, but it is there. I am updated the docs to reflect this.

Reply:
Please see: https://github.com/fir3pho3nixx/Windsor/pull/25#pullrequestreview-131406376

Reply:
> I assume you weren't able to just remove the DisposalConcern from ComponentModel.Lifecycle in CrossWiringComponentModelContributor?

This has now also been done.